File: question_generator.py
from lang_module.openai_client import get_openai_client

def generate_questions(essay_text, target_language, num_questions):
    client = get_openai_client()
    
    system_prompt = f"""
    You are an exam administrator evaluating whether a student has authentically engaged with their written work by generating insightful, personalized questions based on the essay. Focus on the Human Dimension of Fink's model. Use key quotes and make the user reflect. Only use one question mark per question.
    Generate {num_questions} questions.
    Format:
    Question 1: <text>
    ...
    """

    response = client.chat.completions.create(
        model="gpt-4o",
        messages=[
            {"role": "system", "content": system_prompt},
            {"role": "user", "content": essay_text}
        ],
        temperature=0.7,
        max_tokens=1000
    )

    questions_text = response.choices[0].message.content
    questions = []

    for line in questions_text.strip().split("\n"):
        if line.strip().startswith("Question"):
            parts = line.split(":", 1)
            if len(parts) > 1:
                questions.append(parts[1].strip())

    return questions

File: question_generator.ts
import OpenAI from "openai";

const client = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY, // set this in your .env.local
});

export type ComprehensionQuestion = {
  question: string;
  options?: string[];
  correct?: string;
  type?: string;        // e.g., "Multiple Choice", "Short Answer"
  difficulty?: string;  // e.g., "Beginner", "Intermediate"
};

export async function generateQuestions(
  text: string,
  n_questions: number = 5,
  level: string = "beginner",
  model: string = "gpt-4o-mini"
): Promise<string> {
  const prompt = `
    You are a language instructor in the user's target language.
    Take the following text: """${text}""" and generate ${n_questions} short comprehension questions in the same language.
    Generate at least one of each of the following: multiple choice, short answer, sequencing, fill-in-the-blank, open-ended.
    Cover literal (fact recall), inferential (cause-effect), and evaluative (judgment).
    Label each question with difficulty; most should be ${level}, with a few one level higher.
    Present the questions as a numbered list, clearly labeling the type in parentheses (e.g., "(Multiple Choice)").
  `;

  const response = await client.chat.completions.create({
    model,
    messages: [
      { role: "system", content: "You are a helpful assistant." },
      { role: "user", content: prompt },
    ],
    temperature: 0.7,
  });

  return response.choices[0].message?.content ?? "";
}
